docs/infrastructure/administration.md

```diff
@@ -0,0 +1,393 @@
+[rds ssl docs]: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_MySQL.html#MySQL.Concepts.SSLSupport
+
+### Resetting dev/stage to a recent production snapshot
+
+The `treeherder-dev` and `treeherder-stage` RDS instances do not have regular user/sheriff
+interactions that provide certain sets of data (for example classifications of failures).
+
+As such it can be useful to periodically reset them using the latest daily production snapshot.
+This has to be performed by an admin of the `moz-devservices` AWS account, and can be requested
+by filing a bug similar to [bug 1469837].
+
+After an instance is reset:
+
+- The password for the MySQL user `th_admin` must be updated to match the one in `DATABASE_URL`
+  for that Heroku app (since the production password will have been inherited from the snapshot,
+  which won't match). See [Changing MySQL passwords].
+- [Perform a deployment] of the Heroku app, which will ensure that the Django migration and
+  code state are in sync (in case `master` has new migrations not yet seen on production).
+- [Reset the app's Redis cache]
+
+[bug 1469837]: https://bugzilla.mozilla.org/show_bug.cgi?id=1469837
+[changing mysql passwords]: #changing-mysql-passwords
+[perform a deployment]: #deploying-treeherder
+[reset the app's redis cache]: #resetting-the-redis-cache
+
+### Creating a temporary instance
+
+To create a new RDS instance based on the latest daily production DB snapshot:
+
+1. Ensure a bug is filed, which will act as a reminder to delete the instance later.
+2. Go to the [us-east-1 RDS snapshots page].
+3. Select the most recent `treeherder-prod` snapshot.
+4. From the "Actions" menu choose "Restore snapshot".
+5. Set the following options (leaving others at their defaults):
+   - DB instance class: `db.m4.xlarge` (same as dev/stage)
+   - DB instance identifier: `treeherder-dev-bug<NNNNNN>`
+   - Virtual private cloud: `treeherder-prod-vpc`
+   - Subnet group: `treeherder-dbgrp` (press the VPC refresh button to get this to appear)
+   - Public accessibility: Yes
+   - DB parameter group: `treeherder-mysql57`
+6. Select "Restore DB Instance".
+7. Wait 15-20 minutes for the instance to be created and report status as "Available".
+8. Select the instance, click "Modify", enter a randomly generated 30+ character password in
+   the "New master password" field, and press "Continue" to apply the change.
+9. Make a note of the DB hostname recorded under `Endpoint`. It will be in the form
+   `treeherder-dev-bugNNNNNN.HASH.us-east-1.rds.amazonaws.com`.
+
+To use the new instance locally, run this inside the Vagrant shell:
+
+```bash
+export DATABASE_URL='mysql://th_admin:PASSWORD@HOSTNAME/treeherder'
+```
+
+Our Django database configuration automatically enables TLS with non-localhost hostnames.
+
+[us-east-1 rds snapshots page]: https://console.aws.amazon.com/rds/home?region=us-east-1#db-snapshots:
+
+### Granting access to the read-only replica
+
```

Comment:
Is this done by accessing the read-only replica's shell via the mysql workbench connection?

Reply:
Ah good question. 

The `GRANT` SQL statements shown are run like any other manual SQL query - that is using a MySQL client connected to the RDS instance (with one such client being MySQLWorkbench using the steps in the section above this one). However the grant statements have to be run on the master production RDS instance (since the replica doesn't allow writes) - and then the changes will propagate to the replica along with all of the other DB writes.

I've updated the section (locally) to mention this.